backends/p4tools/testgen/core/small_step/table_stepper.cpp

```diff
@@ -0,0 +1,662 @@
+}
+
+const StateVariable& TableStepper::getTableKeyReadVar(const IR::P4Table* table, int keyIdx) {
+    const auto* key = table->getKey()->keyElements.at(keyIdx);
+    return IRUtils::getZombieTableVar(key->expression->type, table, "*keyRead", keyIdx);
+}
+
+const StateVariable& TableStepper::getTableReachedVar(const IR::P4Table* table) {
+    return IRUtils::getZombieTableVar(IR::Type::Boolean::get(), table, "*reached");
+}
+
+// The function compares the right and left expression.
+// Returns 0 if the values are equal, 1 if the left value is greater,
+// and -1 if the right value is greater.
+// If the left value is constant entries and the right value
+// is Mask/Default then priority is given to constants.
+// For more detailed information see
+// https://github.com/p4lang/behavioral-model/blob/main/docs/simple_switch.md#longest-prefix-match-tables
+int TableStepper::compareLPMEntries(const IR::Expression* left, const IR::Expression* right) {
+    const IR::Expression* leftConst = nullptr;
+    const IR::Expression* rightConst = nullptr;
+    if (const auto* leftMask = left->to<IR::Mask>()) {
+        leftConst = leftMask->right;
+    } else {
+        leftConst = left;
+    }
+    if (const auto* rightMask = right->to<IR::Mask>()) {
+        rightConst = rightMask->right;
+    } else {
+        rightConst = right;
+    }
+    if (left->is<IR::Constant>() && right->is<IR::Mask>()) {
+        return 1;
+    }
+
+    if (rightConst->is<IR::DefaultExpression>()) {
+        return 1;
+    }
+    if (leftConst->is<IR::DefaultExpression>()) {
+        return -1;
+    }
+    if ((leftConst->to<IR::Constant>()->value) == (rightConst->to<IR::Constant>()->value)) {
+        return 0;
+    }
+    if ((leftConst->to<IR::Constant>()->value) > (rightConst->to<IR::Constant>()->value)) {
+        return 1;
+    }
+    if ((leftConst->to<IR::Constant>()->value) < (rightConst->to<IR::Constant>()->value)) {
+        return -1;
+    }
+    BUG("Unhandled sort elements type: left: %1% - %2% \n right: %3% - %4% ", left,
+        left->node_type_name(), right, right->node_type_name());
+}
+
+/// Sorts entries for lpm.
+IR::Vector<IR::Entry> TableStepper::sortLpmEntries(const IR::Vector<IR::Entry>& entries) {
+    const IR::Entry* currentEntry = nullptr;
+    IR::Vector<IR::Entry> entriesSorted;
+    for (const auto& entry : entries) {
+        for (const auto& entryKey : entry->keys->components) {
+            if (entryKey->is<IR::Expression>() || entryKey->is<IR::DefaultExpression>() ||
+                entryKey->is<IR::Range>() || entryKey->is<IR::Mask>()) {
+                entriesSorted.push_back(entry);
+            } else {
+                BUG("%1%: Unhandled entry key type: %2%", entryKey, entryKey->node_type_name());
+            }
+        }
+    }
+    const IR::Expression* left = nullptr;
+    const IR::Expression* right = nullptr;
+    for (size_t i = 0; i < entriesSorted.size(); i++) {
```

Comment:
you can't use quicksort?

Reply:
@OleksandrLetychevskyiIntel can you please check this?